### ISSUE TITLE
Lab2 peer review

### ISSUE THREAD

**Issue by federicoboscolo** [FabioSofer/Computational_Intelligence2022#3]
Hi Fabio, I'm going to be reviewing your code for lab2.

First of all, I'm sorry you couldn't find a solution for N=1000, because your solutions for smaller values look very promising.
You might be onto something with your high mutation chance, personally it led me to slightly worse results but everyone's code is different, so it's good that you tried different options and stuck with a choice that wasn't obvious.

I think your fitness function is a bit complex, and maybe it would help a bit to make it simpler as to reduce the computations required and as a result make the program faster. I suggest using tuples to compare different values in one go.

Again, your mutation function is a bit computation-heavy, and usually a mutation is intended as a small change in an individual, so you can try experimenting with that and I'm confident that this along with the fitness is going to bring you results for higher Ns.

Your program structure is solid though, your code makes good use of Pythonisms, and I can tell that you put a lot of work into it. 

In conclusion, I think you can work on this a little bit and it will net you great improvements. Keep coding!
